src/main/venmic.ts

```diff
@@ -69,47 +71,59 @@ function getRendererAudioServicePid() {
 ipcMain.handle(IpcEvents.VIRT_MIC_LIST, () => {
     const audioPid = getRendererAudioServicePid();
 
-    const list = obtainVenmic()
-        ?.list()
-        .filter(s => s["application.process.id"] !== audioPid)
-        .map(s => s["application.name"]);
-
-    const uniqueTargets = [...new Set(list)];
+    const targets = obtainVenmic()
+        ?.list(Settings.store.audioGranularSelect ? ["application.process.id"] : undefined)
+        .filter(s => s["application.process.id"] !== audioPid);
 
-    return list ? { ok: true, targets: uniqueTargets, hasPipewirePulse } : { ok: false, isGlibCxxOutdated };
+    return targets ? { ok: true, targets, hasPipewirePulse } : { ok: false, isGlibCxxOutdated };
 });
 
-ipcMain.handle(IpcEvents.VIRT_MIC_START, (_, targets: string[], workaround?: boolean) => {
+ipcMain.handle(IpcEvents.VIRT_MIC_START, (_, include: Node[]) => {
     const pid = getRendererAudioServicePid();
+    const settings = Settings.store;
 
     const data: LinkData = {
-        include: targets.map(target => ({ key: "application.name", value: target })),
-        exclude: [{ key: "application.process.id", value: pid }]
+        include: include,
+        exclude: [{ "application.process.id": pid }],
+        ignore_devices: settings.audioIgnoreDevices
     };
 
-    if (workaround) {
-        data.workaround = [
-            { key: "application.process.id", value: pid },
-            { key: "media.name", value: "RecordStream" }
-        ];
+    if (settings.audioIgnoreInputMedia ?? true) {
+        data.exclude?.push({ "media.class": "Stream/Input/Audio" });
+    }
+
+    if (settings.audioIgnoreVirtual) {
+        data.exclude?.push({ "node.virtual": "true" });
+    }
+
+    if (settings.audioWorkaround) {
+        data.workaround = [{ "application.process.id": pid, "media.name": "RecordStream" }];
     }
 
     return obtainVenmic()?.link(data);
 });
 
-ipcMain.handle(IpcEvents.VIRT_MIC_START_SYSTEM, (_, workaround?: boolean, onlyDefaultSpeakers?: boolean) => {
+ipcMain.handle(IpcEvents.VIRT_MIC_START_SYSTEM, (_, exclude: Node[]) => {
     const pid = getRendererAudioServicePid();
+    const settings = Settings.store;
```

Comment:
why not use destructuring?
```suggestion
    const { audioOnlySpeakers, audioIgnoreDevices, ... } = Settings.store;
```